lib/models/channel_model.dart

```diff
@@ -41,9 +43,10 @@ class ChannelModel extends ChangeNotifier {
     return this;
   }
 
-  bool handleNotifications(Event notification) {
-    channel!.send('run', notification.toMap());
-    return true;
+  Future sendEvent(Event notification) {
+    final completer = Completer();
+    channel!.send('run', notification.toMap())!.receive("ok", (response) => completer.complete(response));
```

Comment:
the `.receive("error", (error) => completer.completeError(error))` is missing to handle the error return